game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java

```diff
@@ -99,66 +91,54 @@ private void createComponents() {
         .getColumnModel()
         .getColumn(gameTableModel.getColumnIndex(LobbyGameTableModel.Column.Host))
         .setPreferredWidth(67);
-  }
 
-  private void layoutComponents() {
     final JScrollPane scroll = new JScrollPane(gameTable);
     setLayout(new BorderLayout());
     add(scroll, BorderLayout.CENTER);
     final JToolBar toolBar = new JToolBar();
     toolBar.add(hostGame);
     toolBar.add(joinGame);
-    if (isAdmin()) {
+    if (lobbyClient.isAdmin()) {
       toolBar.add(bootGame);
     }
     toolBar.setFloatable(false);
     add(toolBar, BorderLayout.SOUTH);
-  }
 
-  private boolean isAdmin() {
-    return ((IModeratorController) messengers.getRemote(IModeratorController.REMOTE_NAME))
-        .isAdmin();
-  }
-
-  private void setupListeners() {
     hostGame.addActionListener(e -> hostGame());
     joinGame.addActionListener(e -> joinGame());
     bootGame.addActionListener(e -> bootGame());
-    gameTable.getSelectionModel().addListSelectionListener(e -> setWidgetActivation());
+    gameTable
+        .getSelectionModel()
+        .addListSelectionListener(
+            e -> {
+              final boolean selected = gameTable.getSelectedRow() >= 0;
+              joinGame.setEnabled(selected);
+            });
     gameTable.addMouseListener(
-        new MouseListener() {
-          @Override
-          public void mouseClicked(final MouseEvent e) {
-            if (e.getClickCount() == 2) {
-              joinGame();
-            }
-            mouseOnGamesList(e);
-          }
-
-          @Override
-          public void mousePressed(final MouseEvent e) {
-            // right clicks do not 'select' a row by default. so force a row selection at the mouse
-            // point.
-            final int r = gameTable.rowAtPoint(e.getPoint());
-            if (r >= 0 && r < gameTable.getRowCount()) {
-              gameTable.setRowSelectionInterval(r, r);
-            } else {
-              gameTable.clearSelection();
-            }
-            mouseOnGamesList(e);
-          }
-
-          @Override
-          public void mouseReleased(final MouseEvent e) {
-            mouseOnGamesList(e);
-          }
+        new MouseListenerBuilder()
```

Comment:
Using the new MouseListenerBuilder here to clean up the mouse listener code. The implementation bodies were moved to private helper methods.